pallets/automation-time/src/lib.rs

```diff
@@ -525,6 +653,29 @@ pub mod pallet {
 			Ok(task_id)
 		}
 
+		pub fn validate_and_schedule_task(
+			action: Action<T>,
+			who: T:: AccountId,
+			provided_id: Vec<u8>,
+			time: UnixTime,
+		) -> Result<(), Error<T>> {
+			if provided_id.len() == 0 {
+				Err(Error::<T>::EmptyProvidedId)?
+			}
+			Self::is_valid_time(time)?;
+
+			let task_id = Self::schedule_task(who.clone(), provided_id.clone(), time)?;
+			let task: Task<T> = match action {
+				Action::Notify { message } => Task::<T>::create_event_task(who.clone(), provided_id, time, message),
```

Comment:
not sure how to pass in action, cause can't use by reference (cannot pass by reference for task creation). And can't use .clone() for some reason?

Reply:
you can just create the struct here, ie dont use the helpers. I really just needed the helper for the tests. I couldn;t figure out how else to make the tasks in the. test file.